File: quant/src/hiveflow/contracts/cli_output.py
from datetime import datetime, timezone


def _now_iso() -> str:
    return datetime.now(timezone.utc).isoformat()


def ok_output(command: str, run_id: str, data: dict, warnings: list | None = None) -> dict:
    return {
        "schema_version": "1.0.0",
        "command": command,
        "run_id": run_id,
        "status": "ok",
        "generated_at": _now_iso(),
        "source": "system",
        "advice_only": False,
        "decision_weight": 1,
        "data": data,
        "warnings": warnings or [],
        "errors": [],
    }

File: cli/src/contracts/cli_output.rs
use serde::{Deserialize, Serialize};
use serde_json::Value;

#[derive(Debug, Serialize, Deserialize)]
pub struct CliOutput {
    pub schema_version: String,
    pub command: String,
    pub run_id: String,
    pub status: String,
    pub generated_at: String,
    pub source: String,
    pub advice_only: bool,
    pub decision_weight: i64,
    pub data: Value,
    pub warnings: Vec<Value>,
    pub errors: Vec<Value>,
}
